### PR TITLE
Improve error message if `link_patterns` forgotten (#428)

### PR DESCRIPTION
**Pull request by Crozzers**
Added an assertion to spit out a more descriptive error if `link-patterns` is not given a value.

Not sure how to add a test case for this though.

### CODE REVIEW

**Comment by nicholasserra** [comment]
I think i'd rather go a different direction on this one. 

My preference would be to check for the link patterns extra in the Markdown init, and if it's enabled, check that link patterns argument is not None. `raise MarkdownError` if the condition isn't met. 

Thoughts?

**Comment by Crozzers** [comment]
The check has been moved to `Markdown.__init__`. It now checks whether `link-patterns` is in `self.extras` and raises a `MarkdownError` if it is None

File: lib/markdown2.py [hunk 1]
```diff
@@ -248,6 +248,9 @@ def __init__(self, html4tags=False, tab_width=4, safe_mode=None,
                 self._toc_depth = self.extras["toc"].get("depth", 6)
         self._instance_extras = self.extras.copy()
 
+        if 'link-patterns' in self.extras:
+            if link_patterns is None and self.extras['link-patterns'] is None:
```
Comment (nicholasserra) [inline lib/markdown2.py:252]: What's up with the additional  `and self.extras['link-patterns'] is None`? I don't think passing the link patterns is supported through the extras as a dict. Or is this future proofing? Or am I totally missing something :P

Comment (Crozzers) [reply lib/markdown2.py:252]: I assumed that was a supported use case after this was given as the use case in #430 

> ```python
> content = markdown2.markdown( content, extras=["fenced-code-blocks", "tables", "link-patterns"])
> ```

If that is not the case I can remove the extra `and self.extras['link-patterns'] is None`

Comment (nicholasserra) [reply lib/markdown2.py:252]: Yup I think you can nuke it, as nothing is passed in through the extras kwarg.

**Pull request merged**